### ISSUE TITLE
Simualtion taking very long time

### ISSUE THREAD

**Issue by Biswajitks1** [ghdl/ghdl#2732]
We have a very big design nearly 100 files to analyze. When we attempted to run the simulation, it quit before executing the run simulation command of ghdl. By using GDB(GNU project debugger) it came to knew that there is issue with the stack size of system when the stack size of system was changed to unlimited the ghdl simulation did run but it takes around 3 hours of time to do the simulation. While the Xilinx Vivado does it in few minutes.
Can anyone help me to figure out this?
The ghdl version used is:-

GHDL 5.0.0-dev (4.1.0.r63.g9bcaa92a2.dirty) [Dunoon edition]
 Compiled with GNAT Version: 10.5.0
 llvm 14.0.0 code generator
Written by Tristan Gingold.

We tried to use the mcode ghdl as it has time resolution option but that didn't solve the issue.

**Comment by tgingold** [comment]
Unfortunately, we need a reproducer to investigate this issue.  The description is too vague.

**Comment by Biswajitks1** [comment]
I have grouped all the files in the zip atached. The vhdl_simulation.sh script runs the ghdl in herarchial way. If the ulimit -s unlimited is commented out then the ghdl stops the run command(please refer the terminal picture).
[image](https://github.com/user-attachments/assets/255473c4-d435-470d-bdd1-691a910a5c23)

Explanation:-
The vhdl files define a FPGA fabric of the [FABulous framework](https://github.com/FPGA-Research-Manchester/FABulous). The user design is a sequential counter and the generated bitstream is converted to hex and loaded in the testbench the output generated by FPGA and user_design is compared.
[test_files.zip](https://github.com/user-attachments/files/16824460/test_files.zip)

**Comment by tgingold** [comment]
I can run the design.
You have a very large number of processes (~800K), and currently there is no inter-process optimizations.
I will have a closer look on how to deal with such a design.

**Comment by Biswajitks1** [comment]
Did the run complete? I mean how much time did it take to run the design.
Thank you for looking into this.

**Comment by tgingold** [comment]
No, I have stopped it before the end as it progressed too slowly.

**Comment by Biswajitks1** [comment]
Hi, I did some observations on RAM consumption.The RAM consumption is not exceeding more than 1.3GB. I don't know whether this is a issue or not.

**Comment by tgingold** [comment]
No, that should be OK.  It would be nice if it were lower.

**Comment by Biswajitks1** [comment]
Hi, were you able to identify the issue? I was thinking is there any way that we can say ghdl to use more resources and can that reduce simulation time?

**Comment by tgingold** [comment]
I haven't worked on it yet.  It's a long time project.

**Comment by Biswajitks1** [comment]
This issue has been resolved. It was related to the clock speed, and @EverythingElseWasAlreadyTaken solved the issue. He may be able to share more details or other possible causes.
